yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -303,6 +326,59 @@ private[spark] class Client(
   }
 
   /**
+   * Create an archive with the Hadoop config files for distribution.
+   *
+   * These are only really used by the AM, since executors will use the configuration object
+   * broadcast by the driver. But this is the easiest way to make sure the files are available for
```

Comment:
So is this comment true, is it only really needed by the AM or do we need it on the executors for things like hive-site.xml?  If its true then we should change our call to ClientDistributedCacheManager.addResource to pass in the flag to make it appMasterOnly.